src/index.js

```diff
@@ -1,59 +1,96 @@
-import qsm from 'query-string-manipulator'
+const EXPANSION_REGEX = /(?<=\{)[^{]*(?=\})/g
 
-const HAL_EXPANSION_TOKEN = '{?'
-const HAL_CONTINUATION_TOKEN = '{&'
-const tokens = [HAL_EXPANSION_TOKEN, HAL_CONTINUATION_TOKEN]
+const DEFAULT_OPERATOR =
+  { operator: '', separator: ',', named: false, ifEmpty: false }
+const OPERATORS = [
+  { operator: '#', separator: ',', named: false, ifEmpty: false },
+  { operator: '.', separator: '.', named: false, ifEmpty: false },
+  { operator: '/', separator: '/', named: false, ifEmpty: false },
+  { operator: ';', separator: ';', named: true, ifEmpty: false },
+  { operator: '?', separator: '&', named: true, ifEmpty: true },
+  { operator: '&', separator: '&', named: true, ifEmpty: true }
+]
 
-function getOptionsToken (link) {
-  for (const token of tokens) {
-    const index = link.indexOf(token)
-    if (index !== -1) {
-      return [token, index]
-    }
+function getNamedOperatorIfNeeded (operator, varName, el) {
+  if (operator.named) {
+    return `${varName}=${el}`
+  }
+  return el
+}
+
+function getValueFor (args, varName, operator) {
+  if (args[varName] instanceof Array) {
+    return args[varName]
+      .flatMap(el => getNamedOperatorIfNeeded(operator, varName, el))
+      .join(operator.separator)
+  }
+  return getNamedOperatorIfNeeded(operator, varName, [`${args[varName]}`]
+    .join(operator.separator))
+}
+
+function containsArgument (args, v, canBeEmpty) {
+  if (canBeEmpty) {
+    return !!args && typeof args[v] !== 'undefined'
   }
-  return [null, 0]
+  return !!args && !!args[v]
```

Comment:
Please use `Boolean(args && args[v])`. It looks less hacky than `!!` :-)

Reply:
Done